host_capabilities/hostcaller_native.go

```diff
@@ -0,0 +1,60 @@
+//go:build !wasi
+// +build !wasi
+
+// note well: we have to use the tinygo wasi target, because the wasm one is
+// meant to be used inside of the browser
+
+package host_capabilities
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+type MockWapcClient struct {
+	err             error
+	payloadResponse []byte
+}
+
+// Create a new MockWapcClient that simulates a host failing
+// whenever one of its capabilities is invoked by the policy.
+// The error returned by the host is the one provided at construction time
+func NewFailingMockWapcClient(err error) *MockWapcClient {
+	return &MockWapcClient{
+		payloadResponse: []byte{},
+		err:             err,
+	}
+}
+
+// Create a new MockWapcClient that simulates a host successfully
+// completing a request made by the policy.
+// The response is going to be the `responseObj` serialized to JSON.
+// Use the right response type object that is defined inside of the `types.go`
+// file of this package.
+//
+// This function can fail if the `responseObj` provided by the user cannot be
+// encoded to JSON.
+func NewSuccessfulMockWapcClient(responseObj interface{}) (*MockWapcClient, error) {
+	payload, err := json.Marshal(responseObj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MockWapcClient{
+		payloadResponse: payload,
+		err:             nil,
+	}, nil
+}
+
+func (m *MockWapcClient) HostCall(binding, namespace, operation string, payload []byte) (response []byte, err error) {
+	return m.payloadResponse, m.err
+}
+
+// NewHost creates a Host that has a mock waPC client.
+func NewHost() Host {
```

Comment:
This is not a blocking comment, I'm fine with merging like it is this is more a personal opinion. 
I would have just panic here if it's called from native saying that just wasm is supported for this function, and let users create client mocks and pass them as parameters for unit testing. 

With this approach some users might not realize they are using a mock. And the mock is very limited, you can't return something based on the input and count how many times a method was called. That's something quite easy to do if you use a mocking framework such as [gomock](https://github.com/golang/mock) or create your own mocks.

Anyway I'm ok with this as Client is an interface, so users can still pass their own mock if they want.

Reply:
Yes, you're right. this is pretty limited right now, I deliberately kept it simple.

I would go ahead with that and evolve the SDK based on user feedback. In the meantime, as you pointed out, the users can bring their own Client since it's an interface

Reply:
ok!